### ISSUE TITLE
Allow option to pass file mode when opening a PR

### ISSUE THREAD

**Issue by LucaLanziani** [gr2m/octokit-create-pull-request#32]
Proposal to pass file mode when opening a PR:

## Use case

PR containing a bash script that should have the executable flag set

## Proposal

Allow the `files` values to be either a string, `null` or an `object`. 

If the value is an object, the object `must` contain the key `content` that can either be a string or null.

The object might contain the `mode` key that will override the default `100644` mode for the specific file.

## Implementation

I've already drafted a possible implementation in case you are willing to accept this change.

https://github.com/LucaLanziani/octokit-create-pull-request/pull/1/files

**Comment by gr2m** [comment]
have you seen https://github.com/gr2m/octokit-create-pull-request/pull/31? It implements the ability to create/update binary files by setting a file to an object with `content` and `encoding` keys.

I think setting a file to `null` is already implemented?

**Comment by LucaLanziani** [comment]
If I understand correctly that PR allows you to attach binary files, I'm just trying to set the executable flag to a bash script for instance. Do  you want me to rebase my PR on top of #31 changes

**Comment by gr2m** [comment]
Ah I see. Yes, if you could rebase your changes and start a PR, we can continue discussing there. Thanks Luca!

**Comment by gr2m** [comment]
Hey @LucaLanziani, if you are anyone else is still interested, I'd be happy to review a PR. Note that I made quite a few changes in the past few days, it might be easiest to recreate the changes based on that. 

I also added instructions to CONTRIBUTING.md on how to record fixtures for the tests.

**Comment by gr2m** [comment]
>  I would open a new issue if you are ok to add this option

Yes, let's open an issue. I'd like to learn more about your use case and see if others have that similar use cases, too
